### PR TITLE
[KOGITO-2946] Maven artifacts should fail on 404

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-2946

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a testcase that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: scripts/update-maven-information.py [hunk 1]
```diff
@@ -85,9 +86,19 @@ def getMD5(service):
...
+    checkUrl(runnerMD5URL)
     runnerMD5=sp.getoutput("curl -s  {}".format(runnerMD5URL))
     return runnerMD5
 
+def checkUrl(url):
+    '''
+    Check url returns 2xx code. 
+    :param url
+    '''
+    resultCode=int(sp.getoutput('curl -I -s -o /dev/null -w "%{}" {}'.format('{http_code}', url)))
```
Comment (Kaitou786) [inline scripts/update-maven-information.py]: Can we do it this way:  `sp.getoutput('curl -I -s -o /dev/null -w "%{0}" {1}'.format('{http_code}', url))` wdyt?

File: scripts/update-maven-information.py [hunk 2]
```diff
@@ -85,9 +86,19 @@ def getMD5(service):
...
+    if resultCode < 200 or resultCode >= 300:
+        raise ValueError('Got http code {} for url {}'.format(resultCode, url))
```
Comment (Kaitou786) [inline scripts/update-maven-information.py]: same here.

File: scripts/update-maven-information.py (hunk 1 above)
Comment (radtriste) [reply scripts/update-maven-information.py]: updated

File: scripts/update-maven-information.py (hunk 2 above)
Comment (radtriste) [reply scripts/update-maven-information.py]: updated

**Review by Kaitou786 — APPROVED** [review]

**Comment by radtriste** [comment]
Jenkins retest this

**Comment by radtriste** [comment]
Jenkins test this

File: scripts/common.py [hunk 3]
```diff
@@ -258,7 +258,7 @@ def update_examples_ref_in_clone_repo(examples_ref):
     pattern = re.compile(r'(git checkout.*)')
     replacement = "git checkout master"
     if examples_ref != 'master':
-        replacement = "git checkout -b {} {}".format(examples_ref, examples_ref)
+        replacement = "git checkout -b {0} origin/{1}".format(examples_ref, examples_ref)
```
Comment (radtriste) [inline scripts/common.py:261]: just a small fix I detected on the replacement while testing the release pipeline

**Review by spolti — APPROVED** [review]

**Pull request merged**